packages/widget/src/stores/settings/useSettingsStore.ts

```diff
@@ -99,17 +99,38 @@ export const useSettingsStore = createWithEqualityFn<SettingsState>(
             ),
           };
         }),
-      toggleTools: (toolType) =>
+      toggleToolKeys: (toolType, toolKeys) =>
         set((state) => {
-          const enabledTools = { ...state[`_enabled${toolType}`] };
-          const enableAll = Boolean(state[`disabled${toolType}`].length);
-          for (const toolKey in enabledTools) {
-            enabledTools[toolKey] = enableAll;
-          }
+          const allKeysInCollectionEnabled = toolKeys.every(
+            (toolKey) => state[`_enabled${toolType}`][toolKey],
+          );
+
+          // then toggle those keys to false
+          const updatedTools = toolKeys.reduce(
+            (accum, toolKey) => {
+              accum[toolKey] = !allKeysInCollectionEnabled;
+              return accum;
+            },
+            {
+              ...state[`_enabled${toolType}`],
+            },
+          );
+
+          const enableKeys: string[] = [];
+          const disabledKeys: string[] = [];
+
+          Object.entries(updatedTools).forEach(([key, value]) => {
+            if (value) {
+              enableKeys.push(key);
+            } else {
+              disabledKeys.push(key);
+            }
+          });
+
           return {
-            [`_enabled${toolType}`]: enabledTools,
-            [`enabled${toolType}`]: enableAll ? Object.keys(enabledTools) : [],
-            [`disabled${toolType}`]: enableAll ? [] : Object.keys(enabledTools),
+            [`_enabled${toolType}`]: updatedTools,
+            [`enabled${toolType}`]: enableKeys,
+            [`disabled${toolType}`]: disabledKeys,
```

Comment:
Due to these changes here please double-check just in case that we send the correct filtering options to the API.

Reply:
WIll check this in the morning

Reply:
Have check that the bridge and exchanges are passed correctly to the options on request - it looks like its working